types/types.go

```diff
@@ -401,11 +401,17 @@ type ImageInspectInfo struct {
 }
 
 // DockerAuthConfig contains authorization information for connecting to a registry.
+//
+// Deprecated: Use github.com/containers/image/credentials/single's AuthStore instead.
 type DockerAuthConfig struct {
 	Username string
 	Password string
 }
 
+// AuthGetter retrieves credentials for HTTP(S) access.  It returns
+// the username and secret as strings.
+type AuthGetter func(serverURL string) (string, string, error)
```

Comment:
It’s very much not obvious to me that it makes sense to have a single callback for _all kinds of_ “HTTP(S)” access, no matter the protocol on top of HTTPS, or authentication mechanism. (Sure, the `AuthFilePath` name is similarly misleading; we originally thought that it would somehow be generic, but it clearly is not.)

WRT `serverURL`, in docker/distribution the lookup key is not an URL!

Reply:
> WRT `serverURL`, in docker/distribution the lookup key is not an URL!

The interface I'm implementing is [`credentials.Helper`][1], which uses `serverURL`.  This getter is the `Get` method from that interface.  I'm fine using a different string if you want for the input argument (personally, I prefer "[authority][2]", or adjusting the docstring to whatever you want me to use instead, just tell me what to paste in ;).

> ... we originally thought that it would somehow be generic, but it clearly is not...

HTTP(S) auth [is a generic thing][3].  Unfortunately, the Docker APIs don't really conform, since they don't provide a way to feed in a challenge string.  Should we come up with our own API (or lean on one that already exists in Go) here, and then special-case the Docker approach with dummy challenge?

[1]: https://godoc.org/github.com/docker/docker-credential-helpers/credentials#Helper
[2]: https://tools.ietf.org/html/rfc3986#section-3.2
[3]: https://tools.ietf.org/html/rfc7235